easybuild/toolchains/compiler/inteliccifort.py

```diff
@@ -71,13 +71,11 @@ class IntelIccIfort(Compiler):
 
     # used when 'optarch' toolchain option is enabled (and --optarch is not specified)
     COMPILER_OPTIMAL_ARCHITECTURE_OPTION = {
-        systemtools.INTEL : 'xHost',
-        systemtools.AMD : 'xHost',
+        systemtools.X86_64: 'xHost',
```

Comment:
@boegel Hmm... for ARM, one needs to distinguish by architecture (as both AArch32 and AArch64 are of CPU family `ARM`), for X64_64 you want a distinction by CPU family.  I can see various ways to address this:
 - Use the architecture as dictionary key (like now) and overwrite the value in `_set_optimal_architecture` if needed
 - Do it exactly the other way around (family as key, overwrite on architecture)
 - Use a tuple `(architecture, family)` as dictionary key

If you really want to keep the family distinction, I'm leaning towards the last option.  What do you think?